src/functions-and-arrays.js

```diff
@@ -1,19 +1,131 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(a,b) {
+  if (a > b) {
+    return a;
+  }
+  return b; 
+}
+
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+function findLongestWord(array) {
+  let longestWord = '';
+  if (array == '') {
+    return null;
+  } else {
+    for (let i =0; i < array.length; i++) {
+      if (array[i].length > longestWord.length) {
+        longestWord = array[i]
+      } else {
+        continue;
+      }
+    }
+  }
+  return longestWord;
+}
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(arr) {
+  let result = 0;
+  for (let i=0; i < arr.length; i++) {
+    result += arr[i];
+  }
+  return result;
+}
+
+// Bonus Calculate the sum
+
+function sum(arr) {
+  let result = 0;
+  for (let i=0; i < arr.length; i++) {
+    switch(typeof arr[i]) {
+      case 'number':
+        result += arr[i];
+        break;
+      case 'string':
+        result += arr[i].length;
+        break;
+      case 'boolean':
+        if (arr[i] === true) {
+          result++
+        }
+        break;
+      default:
+        throw new Error("Unsupported data type sir or ma'am");
+    } 
+  }
+  return result;
+}
+
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(arr) {
+  let sum = 0;
+  let size = 0;
+  if (arr == '') {
+    return null;
+  }
+  for (let i = 0; i < arr.length; i++) {
+    sum += arr[i];
+    size++
+  }
+  return sum/size;
```

Comment:
Instead of having this size variable you can instead divide by the length of the array:
sum/arr.length